### ISSUE TITLE
Azure Cloud Shell improvements

### ISSUE THREAD

**Issue by JasonRShaver** [Azure/azure-cli#3559]
## Login
While users are automatically logged in, the 'az login' command should still be supported, but not default to a device login.  

Currently it does:
```
jason@Azure:~$ az login
To sign in, use a web browser to open the page https://aka.ms/devicelogin and enter the code BN3K7K4R2 to authenticate.
```

I propose

```
jason@Azure:~$ az login
The Azure Cloud Shell has automatically logged you in.
Type "az login --force" to log in anyway

Your current subscription is: Azure CLI Demo with TTL = 2 days
Type "az account list" to show other available subscriptions
```

## Components
We should disable `az component update` if it does not work.  Currently it does:
```
$ az component update
    100% |████████████████████████████████| 71kB 4.0MB/s
    100% |████████████████████████████████| 51kB 5.1MB/s
    100% |████████████████████████████████| 81kB 6.3MB/s
    100% |████████████████████████████████| 92kB 6.2MB/s
    100% |████████████████████████████████| 71kB 6.4MB/s
    100% |████████████████████████████████| 491kB 1.7MB/s
    100% |████████████████████████████████| 51kB 8.1MB/s
    100% |████████████████████████████████| 61kB 7.0MB/s
    100% |████████████████████████████████| 92kB 8.2MB/s
    100% |████████████████████████████████| 286kB 4.8MB/s
    100% |████████████████████████████████| 133kB 6.0MB/s
    100% |████████████████████████████████| 419kB 3.2MB/s
    100% |████████████████████████████████| 143kB 5.8MB/s
  Running setup.py bdist_wheel for azure-datalake-store ... done
  Running setup.py bdist_wheel for cryptography ... done
Permission denied. Run command with --debug for more information.
If executing az with sudo, you may want sudo's -E and -H flags.
```

**Comment by derekbekoe** [comment]
@tjprescott This is cloud shell not interactive. I created a new Cloud Shell label.

**Comment by yugangw-msft** [comment]
this has been fixed #4268
